### PR TITLE
2.5.0

### PR DESCRIPTION
**Pull request by abedra**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -551,19 +552,13 @@ comments.error   # return error, if any
...
-# Tests
-
-You can run tests with 
-
-```ruby
-rake test
-```
-
-If you do send a pull request, please add passing tests for the new feature/fix.
-
```
Comment (ankitdimania-eligible) [inline README.md]: Is this permanently removed?

Comment (ankitdimania-eligible) [inline README.md]: I guess the context is moved to ```Contributing```. Please confirm.

File: lib/eligible/api_resource.rb [hunk 2]
```diff
@@ -1,16 +1,14 @@
 module Eligible
   class APIResource < EligibleObject
-
     def self.class_name
-      self.name.split('::')[-1]
+      name.split('::')[-1]
```
Comment (ankitdimania-eligible) [inline lib/eligible/api_resource.rb]: how about ```.split('::').last```

File: lib/eligible/ticket.rb [hunk 3]
```diff
@@ -1,40 +1,33 @@
...
+      response, api_key = Eligible.request(:post, "/tickets/#{params[:id]}/comments", api_key, params)
+      Util.convert_to_eligible_object(response, api_key)
+    end
 
-      def get(params, api_key=nil)
-        response, api_key = Eligible.request(:get, "/tickets/#{params[:id]}", api_key, params)
-        Util.convert_to_eligible_object(response, api_key)
-      end
+    def self.all
+      response, api_key = Eligible.request(:get, '/tickets', api_key)
```
Comment (ankitdimania-eligible) [inline lib/eligible/ticket.rb]: :+1:

File: test/test_helper.rb [hunk 4]
```diff
@@ -1,3 +1,6 @@
+require 'simplecov'
+SimpleCov.start
```
Comment (ankitdimania-eligible) [inline test/test_helper.rb]: can we please add filter, group and minimum coverage

```
  SimpleCov.start do
    ## remove anything outside lib
    ## it specifically removes .bundle and spec
    add_filter do |src|
      !(src.filename =~ /\/lib\/eligible\//)
    end

    Dir[SimpleCov.root + '/lib/**/*/'].each do |dir|
      add_group File.basename(dir).capitalize, dir
    end

    minimum_coverage 44
  end
```

File: lib/eligible.rb [hunk 5]
```diff
@@ -28,13 +28,15 @@
 require 'eligible/errors/api_error'
 require 'eligible/errors/invalid_request_error'
 
+# rubocop:disable Metrics/ModuleLength, Style/ClassVars
 module Eligible
   @@api_key = nil
   @@test = false
   @@api_base = 'https://gds.eligibleapi.com/v1.1'
   @@api_version = 1.1
```
Comment (ankitdimania-eligible) [inline lib/eligible.rb]: @rod-eligible @hari-eligible are we still on v1.1 for ruby?

File: eligible.gemspec [hunk 6]
```diff
@@ -4,24 +4,26 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'eligible/version'
 
 Gem::Specification.new do |gem|
-  gem.name          = "eligible"
+  gem.name          = 'eligible'
   gem.version       = Eligible::VERSION
```
Comment (ankitdimania-eligible) [inline eligible.gemspec]: is the gem version updated?

File: lib/eligible.rb [hunk 7]
```diff
@@ -28,13 +28,15 @@
 require 'eligible/errors/api_error'
 require 'eligible/errors/invalid_request_error'
 
+# rubocop:disable Metrics/ModuleLength, Style/ClassVars
 module Eligible
   @@api_key = nil
   @@test = false
   @@api_base = 'https://gds.eligibleapi.com/v1.1'
```
Comment (ankitdimania-eligible) [inline lib/eligible.rb]: shall we construct api_base from api_version
```@@api_base = "https://gds.eligibleapi.com/v#{@@api_version}"```

Comment (abedra) [inline lib/eligible.rb]: Probably, yes.

File: lib/eligible/api_resource.rb (hunk 2 above)
Comment (abedra) [inline lib/eligible/api_resource.rb]: Yep, that's a better thing to do here. I was just removing the unneeded self in this case but that change can be added

File: test/test_helper.rb (hunk 4 above)
Comment (abedra) [inline test/test_helper.rb]: We can absolutely add a minimum. I wanted to talk with people before I added that though.

File: README.md (hunk 1 above)
Comment (abedra) [inline README.md]: Yep, that just moved to the Contributing section. It was on its own with no other content.

File: eligible.gemspec (hunk 6 above)
Comment (abedra) [inline eligible.gemspec]: Not yet. That's usually the last thing I do before cutting a release.

**Pull request merged**